tests/unit/src/test/scala/tests/testProvider/DiscoverTestSuitesSuite.scala

```diff
@@ -0,0 +1,116 @@
+package tests
+
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.testProvider.TestSuiteDiscoveryResult
+
+import tests.TestDiscoveryWrapper._
+
+class DiscoverTestSuitesSuite extends BaseLspSuite("discoverTestSuites") {
+
+  test("munit-test-suite") {
+    for {
+      _ <- Future.successful(cleanWorkspace())
+      _ <- initialize(s"""|/metals.json
+                          |{
+                          |  "app": {
+                          |    "libraryDependencies" : [ "org.scalameta::munit:0.7.29", "org.scalatest::scalatest:3.2.1" ],
+                          |    "scalaVersion": "2.13.6"
+                          |  }
+                          |}
+                          |/app/src/main/scala/foo/bar/MyTestSuite.scala
+                          |package foo.bar
+                          |abstract class AbstractTestSuite1 extends munit.FunSuite 
+                          |
+                          |class MunitTestSuite extends AbstractTestSuite1 {
+                          |  test("foo") {}
+                          |}
+                          |""".stripMargin)
+      _ <- server.server.compilations.compilationFinished(
+        workspace.resolve("app/src/main/scala/foo/bar/MyTestSuite.scala")
+      )
+      _ <- server.executeCommand(ServerCommands.CascadeCompile)
+      _ <- server.didOpen("app/src/main/scala/foo/bar/MyTestSuite.scala")
+      _ <- server.didSave("app/src/main/scala/foo/bar/MyTestSuite.scala")(
+        identity
+      )
+      _ <- server.waitFor(TimeUnit.SECONDS.toMillis(10))
+      res <- server.discoverTestSuites(
+        "app/src/main/scala/foo/bar/MyTestSuite.scala"
+      )
+    } yield {
+      val obtained = res.map(TestDiscoveryWrapper(_))
+      val expected = List(
+        TestDiscoveryWrapper(
+          "app",
+          List(
+            PackageWrapper(
+              "foo",
+              List(
+                PackageWrapper(
+                  "bar",
+                  List(
+                    SuiteWrapper(
+                      "foo.bar.MunitTestSuite",
+                      "MunitTestSuite"
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+
+      assertEquals(expected, obtained)
+    }
+  }
+}
+
+final case class TestDiscoveryWrapper(
```

Comment:
Why do you need to wrap it ? Can't you just use TestSuiteDiscoveryResult.TestSuite ? This seems to be duplciating the hierarchy again.

OR you can even have a json result as string to show what the client receives.

Reply:
Because TestSuiteDiscoveryResults are just classes, without `case` which means that they don't have defined equals or toString method. In addition, they contain `java.util.List` because of GSON. 

I wrote that it's a lot of boilerplate, but it was only for quickly developing and testing things. I think that comparing strings should work well, I like this idea.